src/Microsoft.TemplateEngine.Cli/Commands/create/TemplateOption.cs

```diff
@@ -5,24 +5,94 @@
 
 namespace Microsoft.TemplateEngine.Cli.Commands
 {
-    internal class TemplateOption
+    internal class TemplateOption : IEquatable<TemplateOption>
     {
-        private readonly Option _option;
+        private Option _option;
 
         internal TemplateOption(
             CliTemplateParameter parameter,
-            IReadOnlyList<string> aliases)
+            IReadOnlySet<string> aliases)
         {
             TemplateParameter = parameter;
             Aliases = aliases;
-            _option = parameter.GetOption(aliases);
+            _option = TemplateParameter.GetOption(Aliases);
         }
 
         internal CliTemplateParameter TemplateParameter { get; private set; }
 
-        internal IReadOnlyList<string> Aliases { get; private set; }
+        internal IReadOnlySet<string> Aliases { get; private set; }
 
         internal Option Option => _option;
 
+        public bool Equals(TemplateOption? other)
```

Comment:
nit: not sure how likely it is to be called with same instance here - but if it's possible then it might be a good idea to frontload checking for that:

```
if (ReferenceEquals(this, obj))
{
    return true;
}
```